src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -204,8 +204,24 @@ def add_argument(self, param_name, *option_strings, **kwargs):
 
     def update_argument(self, param_name, argtype):
         arg = self.arguments[param_name]
+        CliCommand._resolve_default_value_from_cfg_file(arg, argtype)
         arg.type.update(other=argtype)
 
+    @staticmethod
+    def _resolve_default_value_from_cfg_file(arg, overrides):
+        if 'configured_default' in overrides.settings:
+            def_config = overrides.settings.get('configured_default', None)
+            if arg.type.settings.get('required', False):
+                setattr(arg.type, 'configured_default_applied', True)
+                parts = def_config.split('/')
+                section = 'core' if len(parts) == 1 else parts[0]
+                config_name = parts[-1]
```

Comment:
The 3 lines above should go into `_config.py` in a helper method.
In the PR, similar logic is used in `configure/custom.py`.